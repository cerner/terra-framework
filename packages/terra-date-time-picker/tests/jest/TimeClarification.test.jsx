import React from 'react';
/* eslint-disable import/no-extraneous-dependencies */
import { shallowWithIntl } from 'terra-enzyme-intl';
import TimeClarification from '../../lib/_TimeClarification';

it('should render a default date time picker', () => {
  const datePicker = shallowWithIntl((
    <TimeClarification
      isOpen
      isOffsetButtonHidden
      onDaylightSavingButtonClick={jest.fn()}
      onStandardTimeButtonClick={jest.fn()}
      onOffsetButtonClick={jest.fn()}
      onRequestClose={jest.fn()}
      releaseFocus={jest.fn()}
      requestFocus={jest.fn()}
    />
  ));
  expect(datePicker).toMatchSnapshot();
});

it('should render a disabled time clarification', () => {
  const datePicker = shallowWithIntl((
    <TimeClarification
      isOpen
      isOffsetButtonHidden
      onDaylightSavingButtonClick={jest.fn()}
      onStandardTimeButtonClick={jest.fn()}
      onOffsetButtonClick={jest.fn()}
      onRequestClose={jest.fn()}
      releaseFocus={jest.fn()}
      requestFocus={jest.fn()}
      disabled
    />
  ));
  expect(datePicker).toMatchSnapshot();
});
