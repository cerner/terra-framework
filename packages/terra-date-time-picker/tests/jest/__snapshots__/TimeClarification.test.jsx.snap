// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a default date time picker 1`] = `
<div>
  <AbstractModal
    ariaLabel="Time Clarification"
    classNameModal="time-clarification"
    classNameOverlay={null}
    closeOnEsc={false}
    closeOnOutsideClick={false}
    isFocused={true}
    isFullscreen={false}
    isOpen={true}
    onRequestClose={[MockFunction]}
    role="dialog"
    rootSelector="[data-terra-base]"
    zIndex="9000"
  >
    <div>
      <header
        className="header"
      >
        <h1
          className="title"
        >
          Terra.dateTimePicker.timeClarification.title
        </h1>
      </header>
      <div
        className="body"
      >
        <p>
          Terra.dateTimePicker.timeClarification.message
        </p>
      </div>
      <div
        className="buttons"
      >
        <Button
          className="button-daylight"
          isBlock={false}
          isCompact={false}
          isDisabled={false}
          isIconOnly={false}
          isReversed={false}
          onClick={[Function]}
          text="Terra.dateTimePicker.timeClarification.button.daylightSaving"
          type="button"
          variant="emphasis"
        />
        <Button
          className="button-standard"
          isBlock={false}
          isCompact={false}
          isDisabled={false}
          isIconOnly={false}
          isReversed={false}
          onClick={[Function]}
          text="Terra.dateTimePicker.timeClarification.button.standardTime"
          type="button"
          variant="emphasis"
        />
      </div>
    </div>
  </AbstractModal>
  <Button
    className="button-offset button-offset-hidden"
    isBlock={false}
    isCompact={true}
    isDisabled={false}
    isIconOnly={false}
    isReversed={false}
    onClick={[MockFunction]}
    text=""
    type="button"
    variant="neutral"
  />
</div>
`;

exports[`should render a disabled time clarification 1`] = `
<div>
  <AbstractModal
    ariaLabel="Time Clarification"
    classNameModal="time-clarification"
    classNameOverlay={null}
    closeOnEsc={false}
    closeOnOutsideClick={false}
    isFocused={true}
    isFullscreen={false}
    isOpen={true}
    onRequestClose={[MockFunction]}
    role="dialog"
    rootSelector="[data-terra-base]"
    zIndex="9000"
  >
    <div>
      <header
        className="header"
      >
        <h1
          className="title"
        >
          Terra.dateTimePicker.timeClarification.title
        </h1>
      </header>
      <div
        className="body"
      >
        <p>
          Terra.dateTimePicker.timeClarification.message
        </p>
      </div>
      <div
        className="buttons"
      >
        <Button
          className="button-daylight"
          isBlock={false}
          isCompact={false}
          isDisabled={false}
          isIconOnly={false}
          isReversed={false}
          onClick={[Function]}
          text="Terra.dateTimePicker.timeClarification.button.daylightSaving"
          type="button"
          variant="emphasis"
        />
        <Button
          className="button-standard"
          isBlock={false}
          isCompact={false}
          isDisabled={false}
          isIconOnly={false}
          isReversed={false}
          onClick={[Function]}
          text="Terra.dateTimePicker.timeClarification.button.standardTime"
          type="button"
          variant="emphasis"
        />
      </div>
    </div>
  </AbstractModal>
  <Button
    className="button-offset button-offset-hidden"
    isBlock={false}
    isCompact={true}
    isDisabled={true}
    isIconOnly={false}
    isReversed={false}
    onClick={[MockFunction]}
    text=""
    type="button"
    variant="neutral"
  />
</div>
`;
